### PR TITLE
Upgrade to BpyBuild 0.3.0

### PR DESCRIPTION
**Pull request by StandingPadAnimations**
This PR upgrades the MCprep build system (BpyBuild) to 0.3.0. BpyBuild 0.3.0 itself is almost done, just currently in the final stages of testing (https://github.com/Moo-Ack-Productions/bpy-build/pull/6, separate PR due to some current issues with repository settings).

To install BpyBuild 0.3.0:
```sh
pip install git+https://github.com/Moo-Ack-Productions/bpy-build@v0.3.0
```

# Changes for Developers
*sales pitch time*

All BpyBuild commands have been shortend, including the BpyBuild command itself. Here are some examples of what's changed:
- `bpy-addon-build --during-build dev` -> `bab -b dev`
- `bpy-addon-build -v 3.0` -> `bab -v 3.0`

BpyBuild also now uses argparse for the CLI, which makes the arguments far more reliable in design

The config has also changed. Here's an example of what changes
## pre-0.3.0 syntax
```yaml
# bpy-build.yaml
during_build:
  dev:
    - copy_file("some_file -> destination in the addon folder")
    - [some_command with-args | and-pipes, another_command]
```

## 0.3.0 syntax
```yaml
# bpy-build.yaml
build_actions:
  dev: "dev.py"
```
```py
# dev.py
def main() -> None:
    with open("mcprep_dev.txt", "w") as f:
        f.write("hi guys c:")

if __name__ == "__main__":
    main()
```

All build actions are now Python based, massively expanding what's possible. This means BpyBuild can also be expanded in its feature set in almost any way you want. Some examples of what's possible now with BpyBuild's new build actions:
- Ignore filters by deleting files with an action
- Downloading and inserting external assets
- A "production" build option for including large files
- Using custom analysis scripts 
- etc

If it can be done in Python, it can now be done in BpyBuild, a massive improvement from BpyBuild's old clunky system.

Versions are no longer strings, which means the following syntax is now unsupported:
```yaml
install_versions:
  - '~/some/path/to/blender/3.5`
```

In the future, a new system will be introduced to handle custom config paths

### CODE REVIEW

**Comment by StandingPadAnimations** [comment]
In time for Christmas too!

File: bpy-build.yaml
Comment (TheDuckCow) [inline bpy-build.yaml:16]: Does this still work for 2.80 with this naming convention? If so, no complaints

**Review by TheDuckCow — APPROVED** [review]
One comment but you're approved so you're free to merge when you see fit

**Comment by StandingPadAnimations** [comment]
@TheDuckCow when you get the chance, can you test this on your machine? Especially with more complex arguments like `-v` and whatnot. Want to make sure BpyBuild'a bugs get ironed out first before merging

**Comment by StandingPadAnimations** [comment]
@TheDuckCow remember the annoying issue in #505? That should be resolved now with the new ignore filters action :D

I've also made a separate folder to store all the action scripts so that everything's organized (default scripts have their own subfolder too). Let me know if any changes need to be made.

**Comment by StandingPadAnimations** [comment]
Updated `run_tests.py` to use the new build command, should we pass the `-dbg` flag as well (for BpyBuild) during testing to get debug information, or should we just keep it as it?

**Comment by TheDuckCow** [comment]
Marking myself to re-requesting review so I can try it with the latest changes of the package itself

**Comment by StandingPadAnimations** [comment]
Doesn't work atm with the changes I made earlier today (but that's a BpyBuild side issue)

**Comment by StandingPadAnimations** [comment]
Alright, issues from yesterday seem to have resolved themselves somehow (cosmic bit flip perhaps?), so it should be ready to try

**Comment by TheDuckCow** [comment]
I'll take a look in the next couple days. Slightly wary of rolling out this for a minor release, but I can see how confident I am with testing in the end. I at least really do know what to look for now.

**Comment by StandingPadAnimations** [comment]
FYI I pushed a change to BpyBuild today that changes `filters` to `ignore_filters` (to make it more explicit what it means). In addition, I initially moved the Python based ignore filter (for `mcprep_addon_tracker.json`) to BpyBuild's ignore filters, but I decided to keep the Python based one since we can give a warning if we find a file we don't want to end up in a final build:
image

Perhaps we should rename this to "Blacklisted Files" and maybe even stop the build if we find a blacklisted file, though I think this is sufficient for now.

**Comment by StandingPadAnimations** [comment]
I think we're good with the migration, just need to get BpyBuild's PR merged before we can merge this

File: bpy-build.yaml
Comment (TheDuckCow) [reply bpy-build.yaml:16]: Per [this comment](https://github.com/Moo-Ack-Productions/bpy-build/pull/6#pullrequestreview-1859500659), seems to not work for 2.8 (or 2.80) with the current branch of v0.3.0 of bpy-build

File: bpy-build.yaml
Comment (TheDuckCow) [inline bpy-build.yaml:24]: On further review, I don't think this filter works very well because it just results in the blend files being missing in the final archive right?

I was most of the way through implementing a "fast" build option which set the ignore filter to be the mcprep_resources folder, and that seemed to be working fine until I realize we fully delete the existing install always via the line:

```
if addon_path.exists():
    shutil.rmtree(addon_path)
```

Not having any of the source files means test fail, so not very useful for dev building. I'd remove this until we have a proper fast flag that can leave some files in place.

**Review by TheDuckCow — APPROVED** [review]
Approving, but let's remove the ignore filter usage from the dev mode (it's not even that much faster unless you skipp the entire resource folder, because much of the time is coming from zipping and unzipping all of the texture files).

**Comment by StandingPadAnimations** [comment]
Assuming there's no more issues, I'll merge this. BpyBuild v0.3.0 is also now availible on PyPI

**Pull request merged**
